jupyter-extension/jupyterlab_pachyderm/pps_client.py

```diff
@@ -29,38 +29,36 @@ async def generate(self, path, config):
         pipeline_spec = create_pipeline_spec("test_pipeline", "python:3.10", dict(pfs=dict(repo="data")), script_name)
         return json.dumps(pipeline_spec)
 
-    async def create(self, path, config):
+    async def create(self, path: str, config: dict):
         """Creates the pipeline from the Notebook file specified.
 
         Args:
             path: The path (within Jupyter) to the Notebook file.
             config: The PPS configuration for the Notebook file.
         """
         get_logger().debug(f"path: {path} | body: {config}")
-        input_spec = config.get("input")
-        pipeline_name = config.get("pipeline_name")
+
+        # TODO: There's a lot of data-munging here that can error.
+        input_spec = config['run'].pop("input")  # Input not allowed in SAME metadata.
+        # Paths are relative to the config file. Since the config is being written into
+        #  a temporary file, we need to specify an absolute path.
+        config['notebook']['path'] = os.path.join(os.getcwd(), os.path.relpath(path, '/'))
+        config['notebook']['name'] = os.path.basename(path)
+
+        if 'requirements' in config['notebook'] and not config['notebook']['requirements']:
+            # Remove requirements field if none are specified.
+            del config['notebook']['requirements']
 
         with NamedTemporaryFile() as temp_config:
-            Path(temp_config.name).write_text(
-                "apiVersion: sameproject.ml/v1alpha1\n"
-                "metadata:\n"
-                f"  name: {pipeline_name}\n"
-                f"  version: 0.0.1\n"
-                "environments:\n"
-                "  default:\n"
-                "    image_tag: combinatorml/jupyterlab-tensorflow-opencv:0.9\n"
-                "notebook:\n"
-                f"  name: TestNotebook\n"
-                f"  path: {Path(__file__).parent}/tests/data/TestNotebook.ipynb\n"
-                # f"  name: {os.path.basename(path)}\n"
-                # f"  path: {path}\n"
-            )
-
-            call = subprocess.run(
-                [sys.executable, "-m", "sameproject.main", "run", "--same-file", temp_config.name, "--target", "pachyderm", "--input", json.dumps(input_spec)],
-                check=True
-            )
-        return json.dumps(dict())
+            Path(temp_config.name).write_text(json.dumps(config))
+            command = [sys.executable, "-m", "sameproject.main", "run"]
+            command.extend(["--same-file", temp_config.name])
+            command.extend(["--target", "pachyderm"])
+            command.extend(["--input", input_spec])
+            call = subprocess.run(command, capture_output=True)
+            error = call.stderr.decode() if call.returncode else None
+
+        return json.dumps(dict(error=error))
```

Comment:
Currently this is always replying with a 200 code. @smalyala Is this what the mount extension does or is there a way to return specific error codes?